runtime/lua/vim/treesitter/query.lua

```diff
@@ -692,10 +692,14 @@ function Query:iter_captures(node, source, start, stop)
   if type(source) == 'number' and source == 0 then
     source = api.nvim_get_current_buf()
   end
+  local bufnr = source --[[@as integer]]
+  if type(source) == 'string' then
+    bufnr = api.nvim_get_current_buf()
```

Comment:
I wonder what is the logical meaning of this. Why do we have the current buffer here, and why `_rawquery` does have to take `bufnr` as input? I guessed this is in order to get the value of the option associated with the buffer, right?

If the source string is not associated with any buffer, can we use the global value of `&tslm` instead of the buffer-local value?

Reply:
I needed `bufnr` to make the buffer local option work, but I guess I could do what you said and pass along the string values to the c code. I think @clason's suggestion below might make more sense though.